### PR TITLE
E2E Tests: Add link to E2E dashboard to the output of the runner.

### PR DESCRIPTION
**Pull request by dereksmart**
The visual E2E reports are nice to look at. This PR attempts to add a link as part of the test runner's output, so one could discover on a failing job by 
- clicking `details` in the PR's failing status check
- goes to the failing test job
- will see the link there.

I considered finding a way to override the `details` link in the status checks themselves, but bailed because I think there's value in the first link going to a place that you can re-run the tests easily. 

For example, [this job](https://github.com/Automattic/jetpack/runs/5425798836?check_suite_focus=true) would echo a link to [this report](https://automattic.github.io/jetpack-e2e-reports/22826/report/). 

#### Changes proposed in this Pull Request:

Adds a link to E2E dashboard in test logs. 

#### Jetpack product discussion

N/A

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

Look at the test runs of this PR?

### CODE REVIEW

**Comment by dereksmart** [comment]
[Almost worked](https://github.com/Automattic/jetpack/runs/5426533672?check_suite_focus=true#step:6:30)! 

Just need to figure out how to properly get that variable in there...

**Comment by brbrr** [comment]
FWIW, the report link is easily accessible from gh-actions bot comment above: https://github.com/Automattic/jetpack/pull/23249#issuecomment-1059412095

File: .github/workflows/e2e-tests.yml [hunk 1]
```diff
@@ -119,7 +119,9 @@ jobs:
       working-directory: ${{ matrix.path }}
       env:
         JSON_ARGS: ${{ toJSON(matrix.testArgs) }}
+        PR_NUMBER: ${{ github.event.pull_request.number }}
       run: |
+        echo "View results on E2E dashboard: https://automattic.github.io/jetpack-e2e-reports/${PR_NUMBER}/report/" &&
         mapfile -t TEST_ARGS < <(jq -r '.[]' <<<"$JSON_ARGS")
         pnpm run test:run -- "${TEST_ARGS[@]}"
```
Comment (adimoldovan) [inline .github/workflows/e2e-tests.yml:126]: How about:

```suggestion
        mapfile -t TEST_ARGS < <(jq -r '.[]' <<<"$JSON_ARGS")
        echo "::group::Tests output"
        EXIT_CODE=0
        pnpm run test:run -- "${TEST_ARGS[@]}" || EXIT_CODE=$?
        echo "::endgroup::"
        echo "View results on E2E dashboard: https://automattic.github.io/jetpack-e2e-reports/${PR_NUMBER}/report"
        exit $EXIT_CODE
```

Comment (dereksmart) [reply .github/workflows/e2e-tests.yml:126]: Nice, thanks!

**Comment by adimoldovan** [comment]
PR_NUMBER will be undefined for merges to master. The report URL for those is "https://automattic.github.io/jetpack-e2e-reports/master/report". 
But, there's more:
- for "atomic" scheduled test runs we're using [atomic report](https://automattic.github.io/jetpack-e2e-reports/atomic/report)
- for scheduled blocks tests with Gutenberg latest version we're using [gutenberg](https://automattic.github.io/jetpack-e2e-reports/gutenberg/report) report.

To simplify:
- if needs.create-test-matrix.outputs.runName is defined, the URL will be `needs.create-test-matrix.outputs.runName/report`
- if needs.create-test-matrix.outputs.runName is undefined and event is pull_request the URL will be `pr_number/report`
- if needs.create-test-matrix.outputs.runName is undefined and event is push the URL will be `branch/report`

**Pull request closed without merging**